.github/workflows/generate-schema.yaml

```diff
@@ -0,0 +1,34 @@
+name: Generate Supergraph Schema
+
+on:
+  workflow_call:
+
+jobs:
+  generate_schema:
+    # Deduplicate jobs from pull requests and branch pushes within the same repo.
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: ubuntu-latest
+    env:
+      ARGO_SERVER_SCHEMA_URL: https://raw.githubusercontent.com/argoproj/argo-workflows/main/api/jsonschema/schema.json
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4.1.7
+
+      - name: Install stable toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1.10.0
+
+      - name: Cache Rust Build
+        uses: Swatinem/rust-cache@v2.7.3
+
+      - name: Generate Schema
+        run: >
+          cargo run
+          schema
+          --path workflows.graphql
+        working-directory: graph-proxy
+
+      - name: Upload Schema Artifact
+        uses: actions/upload-artifact@v4.4.0
+        with:
+          name: graphql-schema
+          path: ./graph-proxy/workflows.graphql
```

Comment:
I think this should be taken out and only present in the "Graph Proxy Schema" workflow

Reply:
Since we generate and share the schema with `compose_supergaph` job, we need it there!

Reply:
Can we not move this `step` into that `job`?